app/src/main/java/org/wikipedia/util/ResourceUtil.kt

```diff
@@ -0,0 +1,72 @@
+package org.wikipedia.util
+
+import android.content.ContentResolver
+import android.content.Context
+import android.content.res.ColorStateList
+import android.content.res.Resources.NotFoundException
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.net.Uri
+import android.util.TypedValue
+import android.view.MenuItem
+import androidx.annotation.*
+import androidx.appcompat.content.res.AppCompatResources
+import androidx.core.content.ContextCompat
+import androidx.core.graphics.drawable.DrawableCompat
+import androidx.core.view.MenuItemCompat
+
+object ResourceUtil {
+
+    @JvmStatic
+    fun bitmapFromVectorDrawable(context: Context, @DrawableRes id: Int, @ColorRes tintColor: Int?): Bitmap {
+        val vectorDrawable = AppCompatResources.getDrawable(context, id)
+        val width = vectorDrawable!!.intrinsicWidth
+        val height = vectorDrawable.intrinsicHeight
+        vectorDrawable.setBounds(0, 0, width, height)
+        if (tintColor != null) {
+            DrawableCompat.setTint(vectorDrawable, ContextCompat.getColor(context, tintColor))
+        }
+        val bm = Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888)
+        val canvas = Canvas(bm)
+        vectorDrawable.draw(canvas)
+        return bm
+    }
+
+    private fun getThemedAttribute(context: Context, @AttrRes id: Int): TypedValue? {
+        val typedValue = TypedValue()
+        return if (context.theme.resolveAttribute(id, typedValue, true)) {
+            typedValue
+        } else null
+    }
+
+    @JvmStatic
+    @AnyRes
+    fun getThemedAttributeId(context: Context, @AttrRes id: Int): Int {
+        val typedValue = getThemedAttribute(context, id)
+                ?: throw IllegalArgumentException("Attribute not found; ID=$id")
+        return typedValue.resourceId
+    }
+
+    @JvmStatic
+    @ColorInt
+    fun getThemedColor(context: Context, @AttrRes id: Int): Int {
+        val typedValue = getThemedAttribute(context, id)
+                ?: throw IllegalArgumentException("Attribute not found; ID=$id")
+        return typedValue.data
+    }
+
+    @Throws(NotFoundException::class)
```

Comment:
Would it be possible to import ` kotlin.jvm.Throws` to make this correctly?

Reply:
done @cooltey